v0/destinations/pinterestConversion/transform.js

```diff
@@ -37,9 +38,17 @@ const responseBuilderSimple = finalPayload => {
 
 const identifyPageResponseBuilder = (message, { Config }) => {
   let processedUserPayload;
-  const { appId, advertiserId } = Config;
+  const { appId, advertiserId, enableDeduplication, deduplicationKey } = Config;
   // ref: https://s.pinimg.com/ct/docs/conversions_api/dist/v3.html
   const processedCommonPayload = processCommonPayload(message);
+  /* 
+    message deduplication facility is provided *only* for the users who are using the *new configuration*.
+    if  "enableDeduplication" is set to *true* and "deduplicationKey" is set via webapp, that key value will be 
+    sent as "event_id". On it's absence it will fallback to "messageId".
+  */
+  if(isDefinedAndNotNull(enableDeduplication) && enableDeduplication === true) {
```

Comment:
Do we need to consider `enableDeduplication`? As we are taking deduplicationKey from the dashboard, depending on the availability of deduplicationKey we can set the `event_id` to `messageId` or `deduplicationKey`.

Reply:
yeah, but same thing is being done in snapPixel conversion, and that is what was suggested to do.

Reply:
We have to do that in snapPixel for backward compatibility as there is a mapping of dedupId from properties.dedupId. WDYT @utsabc ?

Reply:
Ideally we should push users to use the webapp config way to do it. So we can depriciate `properties.dedupId`